Module 2. CRUD/src/routes/users.routes.ts

```diff
@@ -0,0 +1,55 @@
+import { userRequestSchema } from './../models/schemas';
+import express, { Request, Response, Router } from 'express';
+import asyncHandler from 'express-async-handler';
+import { createValidator, ValidatedRequest } from 'express-joi-validation';
+
+import { autoSuggestSchema, userSchema, autoSuggestRequestSchema } from '../models/schemas';
+import { User } from '../models/user.interface';
+import { UserService } from './../services/users.service';
+
+const router: Router = express.Router();
+const validator = createValidator();
+const userService = new UserService;
+
+router.get('/', validator.query(autoSuggestSchema),
+    asyncHandler(async (req: ValidatedRequest<autoSuggestRequestSchema>, res: Response) => {
+        const { loginSubstring, limit } = req.query;
+        if (loginSubstring) {
+            await userService.getAutoSuggestUsers(loginSubstring, limit || 2)
+            .then(user => res.json(user));
```

Comment:
Please don't use mix of async/await and promise styles.
const users = await userService...
return res.json(users)